### PR TITLE
switched simulation from fixed fragment size to distribution

### PR DESCRIPTION
**Pull request by mschilli87**
should work as is, but changing `f_size_sim = np.array([400])` to `f_size_sim = np.array([399,400])` and
`f_prob_sim = np.array([1])` to `f_prob_sim = np.array([.1,.9])` results in a shift of  `est_pAlen` from the expected `42` to `43`.

### CODE REVIEW

File: simulate.py [hunk 1]
```diff
@@ -30,5 +39,7 @@ def simulate_reads(genes,pAi,reads_per_gene=100,pAlen=42,fragment_length=400):
         interval=intervals[0] # pick first 3' UTR isoform for now
         reads[gene]=[]
         for read in range(reads_per_gene):
+            r=random.random()
+            fragment_length=f_size[min([i for i in range(len(f_cum)) if r<=f_cum[i]])]
             reads[gene].append(int(interval['start'])-1+pAlen-fragment_length)
```
Comment (mschilli87) [inline simulate.py]: this `-1` here I am not so sure anymore after seeing the weird behaviour ;)

Comment (mschilli87) [inline simulate.py]: Could this be a strand issue? If so, please open an issue so I'll add strand awareness to the simulations.

**Comment by nukappa** [comment]
i checked this also. played with different reads and/or bioanalyzer "profiles" and model behaves as it should... iguess you'll push more things here so i leave it for now?

**Comment by mschilli87** [comment]
Yes, you can leave it around for now. Do you have any clue why the shift to `43` happens?

---

**edit:** Opened #66 to further discuss that issue.

**Comment by mschilli87** [comment]
I merged in current `master` to be able to work on this more efficiently thanks to https://github.com/rajewsky-lab/polyA/commit/0e362b1e04be027d48df75f234c1110a3ea9ce81.

**Comment by mschilli87** [comment]
Simulating random annealing of the oligo-d(T) to the poly(A) breaks the 42 expected value for the fixed 400nt insert size case (I've seen a single 41 in a local run) [breaking the corresponding unit test](https://travis-ci.org/rajewsky-lab/polyA/jobs/138339302#L457) but it seems to greatly improve the 90/10-95/100nt cases (a few 41s and 43s amongs many 42s) as we suspected.
I'll continue with real profiles and so on this branch but I guess we might have to loosen the unit test criterion a bit eventually. Any suggestions? Person correlation (p-value)?

**Comment by nukappa** [comment]
very nice! yes, please continue with real profiles. we'll think about the unit test later..

**Pull request merged**